### PR TITLE
Blog-App: Processing Data In Models

### PR DESCRIPTION
**Pull request by jmoseka**
In this PR, the following methods were created

-  Users
    - A method that returns the 3 most recent posts for a given user.
-  Posts
    - A method that updates the posts counter for a user.
    - A method which returns the 5 most recent comments for a given post.
-  Comments
    - A method that updates the comments counter for a post.
-  Likes

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  belongs_to :user
```
Comment (rloterh) [inline app/models/post.rb]: - [ ] As per the given _ERD_, the _Post_ **association** should have _Post_ belong to _User_ **through** _Author_, an alias of _User_, like below:
`belongs_to :author, class_name: 'User'` 

    Kindly ensure to improve upon this aspect.

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user
```
Comment (rloterh) [inline app/models/comment.rb]: - [ ] As per the given _ERD_, the _Comment_ **association** should have _Comment_ belong to _User_ **through** _Author_, an alias of _User_, like below:
`belongs_to :author, class_name: 'User'` 

    Kindly ensure to improve upon this aspect.

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :user
```
Comment (rloterh) [inline app/models/like.rb]: - [ ] As per the given _ERD_, the _Like_ **association** should have _Like_ belong to _User_ **through** _Author_, an alias of _User_, like below:
`belongs_to :author, class_name: 'User'` 

    Kindly ensure to improve upon this aspect.

File: app/models/post.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
```
Comment (rloterh) [inline app/models/post.rb]: - [ ] OPTIONAL
It is a nice idea to use one of the Rails Active Callbacks `after_save` to call the `update_posts_counter` method every time a new post has been saved to database, then make the `update_posts_counter` method private, similar to:

```ruby
 # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

File: app/models/comment.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+  belongs_to :post
+
+  def update_comments_counter
```
Comment (rloterh) [inline app/models/comment.rb]: - [ ] OPTIONAL
It is a nice idea to use one of the Rails Active Callbacks `after_save` to call the `update_comments_counter method every time a new like has been saved to database, then make the `update_comments_counter` method private, similar to:

```ruby
 # app/models/comments.rb
  after_save :update_comments_counter

  private

  def update_comment_counter
    author.increment!(:comments_counter)
  end
```

File: app/models/like.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+  belongs_to :post
+
+  def update_likes_counter
+    post.increment!(:likesCounter)
+  end
```
Comment (rloterh) [inline app/models/like.rb]: - [ ] OPTIONAL
It is a nice idea to use one of the Rails Active Callbacks `after_save` to call the `update_likes_counter method every time a new like has been saved to database, then make the `update_likes_counter` method private, similar to:

```ruby
 # app/models/likes.rb
  after_save :update_like_counter

  private

  def update_like_counter
    author.increment!(:likes_counter)
  end
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Mila, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Descriptive Pull Request
- Good ReadMe


## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly check the comments marked **[OPTIONAL]** under the review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 

## Hi @jmoseka 👋, 

Well done implementing all the required and even optional changes 👏 
Your project is now complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**